au4/src/projectscene/view/clipsview/timelineruler.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <QtGui/QColor>
+#include <QtGui/QFont>
+#include <QtQuick/QQuickPaintedItem>
+
+#include "rulerformat.h"
+#include "timeformat.h"
+
+class QPainter;
+
+using TickLabelInfo = std::tuple<int, QString, RulerFormat::TickType>;
+using TickLabels = QVector<TickLabelInfo>;
+
+struct TickInfo {
+    double major; // distance between major ticks [s]
+    double minor; // distance between minor ticks [s]
+    double minorMinor; // distance between minorMinor ticks [s]
+    int digits; // number of digits displayed after the decimal point [s]
+};
+
+class TimelineRuler : public QQuickPaintedItem
+{
+    Q_OBJECT
+
+    Q_PROPERTY(int offset READ Offset WRITE SetOffset NOTIFY offsetChanged FINAL)
+    Q_PROPERTY(double zoom READ Zoom WRITE SetZoom NOTIFY zoomChanged FINAL)
+    Q_PROPERTY(QFont textFont READ TextFont WRITE SetTextFont NOTIFY textFontChanged FINAL)
+    Q_PROPERTY(QColor textColor READ TextColor WRITE SetTextColor NOTIFY textColorChanged FINAL)
+    Q_PROPERTY(QColor tickColor READ TickColor WRITE SetTickColor NOTIFY tickColorChanged FINAL)
+
+signals:
+    void offsetChanged();
+    void zoomChanged();
+    void textFontChanged();
+    void textColorChanged();
+    void tickColorChanged();
+
+public:
+    explicit TimelineRuler(QQuickItem* parent = nullptr);
+    virtual ~TimelineRuler() = default;
+
+    void paint(QPainter* painter) override;
+
+    int Offset() const;
+    void SetOffset(int offset);
+
+    double Zoom() const;
+    void SetZoom(double zoom);
+
+    QFont TextFont() const;
+    void SetTextFont(const QFont& font);
+
+    QColor TextColor() const;
+    void SetTextColor(const QColor& color);
+
+    QColor TickColor() const;
+    void SetTickColor(const QColor& color);
+
+protected:
+    void componentComplete() override;
+
+private:
+    void prepareTickData(QVector<QLineF>& ticks, TickLabels& labels, TickInfo& tickInfo, TimeFormat& format, double w, double h);
+    void drawLabels(QPainter* painter, const TickLabels& labels, double w, double h);
+    void drawTicks(QPainter* painter, const QVector<QLineF>& ticks);
+
+    int m_offset { 0 };
+    double m_zoom{ 0 };
```

Comment:
in AU4 we prefer to use `=` for initialization, and only use `{}` where necessary
```
double m_zoom = 0.0;
```

Reply:
Isn't it a good habit to use {} initialization? I mean it has advantage of additional check - it saves you from conversions that lose information like:
int b{8.1} -> will produce compilation error
I know it's rather rare case but :P

Reply:
There may be some benefits to using `{}`, but the big value is readability, and easy to read code means consistent code. It's common practice in the muse framework to use `=`, and there's no good reason to change hundreds of thousands of lines of code to do otherwise :)

More philosophical thoughts, if we use `{}` to initialize class members, then it is consistent to use `{}`  to initialize local variables, but in practice this is not the case, in one place it’s one thing, in another place it’s another :)

Another point, I often see `double v {}`, but this only raises questions about what the default value is, and how much time is spent on thinking depends on the developer’s experience :) In the `double v = 0.0;` option, for developers of any experience the same minimum time of awareness.

i.e., pros and cons of both approaches can be found

If it is important for you to use `{}`, for technical reasons, or you are used to it and think that it is more correct, then I do not insist on using `=`. 
I see that AU3 uses `{}`, so it could be a big cognitive problem to switch to `=` for no obvious reason (other than code consistency)

Reply:
It's not that important for me, I'll switch to `=`, that's not a problem :) It's just the first time I see a different opinion about that, but it's always good to get a different point of view - that's why I asked. Good point with `double v{}`, I didn't think about that